messaging/src/main/java/org/axonframework/eventhandling/AddedTrackerStatus.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling;
+
+/**
+ * References a new {@link EventTrackerStatus} which work just has started on. Can be used as a marker to react on in
+ * the {@link EventTrackerStatusChangeListener}.
+ *
+ * @author Steven van Beelen
+ * @since 4.4
+ */
+public class AddedTrackerStatus extends WrappedTrackerStatus {
```

Comment:
Why introducing `Added` and `Removed``TrackerStatus` and not adding two default methods to the `EventTrackerStatus` interface such as `boolean new()` and `boolean removed()`?

Reply:
I intended to embrace the [special case pattern](https://www.martinfowler.com/eaaCatalog/specialCase.html), which my gut told me would work best with dedicated objects.

I even played with the idea of wrapping the entire collection containing all the status objects, so that it's the collection which knows when something is added, removed or updated. That became rather ugly, rather quickly though, so I abandoned that for the time being.

But, what do you think, should we move away from this special case pattern here?

Reply:
We had a think and chat about it. Adding `default` methods which state whether an `EventTrackerStatus` was added or removed make the API within the `EventTrackerStatusChangeListener` a lot more friendly, as no `instanceof` checks or casts would ever be needed. We thus deduced to introduce these methods to promote a nicer API.